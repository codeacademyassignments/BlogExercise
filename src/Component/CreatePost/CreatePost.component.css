.flex-outer{
  display: flex;
  flex-direction: column;
  /* justify-content: space-around; */
  /* height: 76vw; */
  border: 1px solid black;
  list-style-type: none;
}

.button{
  /* width: 13%;
  height: 28px; */
  border-radius: 10px;
  /* font-size: 121%; */
  transform: scale(2);
  margin:2% 2% 2% 6%;
}