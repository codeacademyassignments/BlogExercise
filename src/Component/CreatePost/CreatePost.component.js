import React from 'react';
import './CreatePost.component.css';
import ListElement from '../ListElement/ListElement.component';

const CreatePost = () => (
  <form>
    <ul className="flex-outer">
      <ListElement label="Date" inputType="text" placeholder="Date of Blog" />
      <ListElement label="Reading Time" inputType="text" placeholder="Reading Time for Blog" />
      <ListElement label="Title" inputType="text" placeholder="Title for Blog" />
      <ListElement label="Description" inputType="text" placeholder="Description for Blog" />
      <li>
        <button type="button" className="button" onClick={() => { window.location.href = 'Home'; }}>Create Post</button>
      </li>
    </ul>
  </form>
);

export default CreatePost;
