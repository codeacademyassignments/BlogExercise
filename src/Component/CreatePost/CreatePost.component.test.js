// import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import React from 'react';
import CreatePost from './CreatePost.component';

describe('CreatePost', () => {
  it('snapshot should match', () => {
    const tree = shallow(<CreatePost />);
    expect(tree).toMatchSnapshot();
  });
});
