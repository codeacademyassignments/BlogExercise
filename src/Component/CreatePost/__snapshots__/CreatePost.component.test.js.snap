// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatePost snapshot should match 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreatePost />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul
        className="flex-outer"
      >
        <ListElement
          inputType="text"
          label="Date"
          placeholder="Date of Blog"
        />
        <ListElement
          inputType="text"
          label="Reading Time"
          placeholder="Reading Time for Blog"
        />
        <ListElement
          inputType="text"
          label="Title"
          placeholder="Title for Blog"
        />
        <ListElement
          inputType="text"
          label="Description"
          placeholder="Description for Blog"
        />
        <li>
          <button
            className="button"
            onClick={[Function]}
            type="button"
          >
            Create Post
          </button>
        </li>
      </ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ListElement
            inputType="text"
            label="Date"
            placeholder="Date of Blog"
          />,
          <ListElement
            inputType="text"
            label="Reading Time"
            placeholder="Reading Time for Blog"
          />,
          <ListElement
            inputType="text"
            label="Title"
            placeholder="Title for Blog"
          />,
          <ListElement
            inputType="text"
            label="Description"
            placeholder="Description for Blog"
          />,
          <li>
            <button
              className="button"
              onClick={[Function]}
              type="button"
            >
              Create Post
            </button>
          </li>,
        ],
        "className": "flex-outer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "inputType": "text",
            "label": "Date",
            "placeholder": "Date of Blog",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "inputType": "text",
            "label": "Reading Time",
            "placeholder": "Reading Time for Blog",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "inputType": "text",
            "label": "Title",
            "placeholder": "Title for Blog",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "inputType": "text",
            "label": "Description",
            "placeholder": "Description for Blog",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="button"
              onClick={[Function]}
              type="button"
            >
              Create Post
            </button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Create Post",
              "className": "button",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "Create Post",
            "type": "button",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul
          className="flex-outer"
        >
          <ListElement
            inputType="text"
            label="Date"
            placeholder="Date of Blog"
          />
          <ListElement
            inputType="text"
            label="Reading Time"
            placeholder="Reading Time for Blog"
          />
          <ListElement
            inputType="text"
            label="Title"
            placeholder="Title for Blog"
          />
          <ListElement
            inputType="text"
            label="Description"
            placeholder="Description for Blog"
          />
          <li>
            <button
              className="button"
              onClick={[Function]}
              type="button"
            >
              Create Post
            </button>
          </li>
        </ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ListElement
              inputType="text"
              label="Date"
              placeholder="Date of Blog"
            />,
            <ListElement
              inputType="text"
              label="Reading Time"
              placeholder="Reading Time for Blog"
            />,
            <ListElement
              inputType="text"
              label="Title"
              placeholder="Title for Blog"
            />,
            <ListElement
              inputType="text"
              label="Description"
              placeholder="Description for Blog"
            />,
            <li>
              <button
                className="button"
                onClick={[Function]}
                type="button"
              >
                Create Post
              </button>
            </li>,
          ],
          "className": "flex-outer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "inputType": "text",
              "label": "Date",
              "placeholder": "Date of Blog",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "inputType": "text",
              "label": "Reading Time",
              "placeholder": "Reading Time for Blog",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "inputType": "text",
              "label": "Title",
              "placeholder": "Title for Blog",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "inputType": "text",
              "label": "Description",
              "placeholder": "Description for Blog",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="button"
                onClick={[Function]}
                type="button"
              >
                Create Post
              </button>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Create Post",
                "className": "button",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": "Create Post",
              "type": "button",
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
